### PR TITLE
fix(LINGUIST-356): Cover edge case where user has no/insufficient active lists/words in WordSelectionService.java

### PR DESCRIPTION
**Pull request by tolgaozgun**

### CODE REVIEW

File: src/main/java/app/linguistai/bmvp/service/wordbank/WordSelectionService.java [hunk 1]
```diff
@@ -116,16 +120,40 @@ private List<UnknownWord> selectWords(UUID userId, int selectSize) throws Except
...
+
+                    selectedWords.add(unknownWord);
+                }
+            });
 
             if (selectedWords.size() != initialSelectSize) {
                 log.error("Error in selecting new words, selected words size is not equal to initial select size, " +
                         "selectedWords size: {}, initialSelectSize: {}", selectedWords.size(), initialSelectSize);
-                throw new NotFoundException("Error in selecting new words, you need to add more unknown words to your word bank");
+                throw new SomethingWentWrongException("Error in selecting new words, you need to add more unknown words to your word bank");
```
Comment (can-aslan) [inline src/main/java/app/linguistai/bmvp/service/wordbank/WordSelectionService.java:151]: Just `You need to add more unknown words to your word bank` could be more user friendly if we are displaying this to the user. You can mark as resolved directly if not.

File: src/main/java/app/linguistai/bmvp/service/wordbank/WordSelectionService.java [hunk 2]
```diff
@@ -116,16 +120,40 @@ private List<UnknownWord> selectWords(UUID userId, int selectSize) throws Except
...
+            if (selectedWords.size() == initialSelectSize) {
+                selectedWords.addAll(unknownWordRepository.findRandomByOwnerListUserIdAndOwnerListIsActiveAndConfidence(
+                        userId, true, confidences, selectSize));
+                return selectedWords;
+            }
+
+            log.error("Defaulting to the predefined word list, as selected words size is not equal to initial select size, " +
+                    "selectedWords size: {}, initialSelectSize: {}", selectedWords.size(), initialSelectSize);
+
+            QPredefinedWordList predefinedWordList = readPredefinedWordListFromYamlFile(DEFAULT_WORD_LIST_FILE);
```
Comment (can-aslan) [inline src/main/java/app/linguistai/bmvp/service/wordbank/WordSelectionService.java:134]: I'm just curious, is reading from the .yaml file each time more performant, or would it be better to store these predefined words in a new DB table and fetch from there?

**Review by can-aslan — COMMENTED** [review]
LGTM, just left some comments, can you check them and ping me again to review?

File: src/main/java/app/linguistai/bmvp/service/wordbank/WordSelectionService.java [hunk 3]
```diff
@@ -116,16 +120,40 @@ private List<UnknownWord> selectWords(UUID userId, int selectSize) throws Except
...
             if (selectedWords.size() != initialSelectSize) {
                 log.error("Error in selecting new words, selected words size is not equal to initial select size, " +
                         "selectedWords size: {}, initialSelectSize: {}", selectedWords.size(), initialSelectSize);
-                throw new NotFoundException("Error in selecting new words, you need to add more unknown words to your word bank");
+                throw new NotFoundException("Error in selecting new words, you need to have at least " + initialSelectSize + " unknown words in your word lists");
             }
 
+            selectedWords.addAll(unknownWordRepository.findRandomByOwnerListUserIdAndOwnerListIsActiveAndConfidence(
+                    userId, true, confidences, selectSize));
+
```
Comment (KardelenCeren) [inline src/main/java/app/linguistai/bmvp/service/wordbank/WordSelectionService.java:156]: what does this call do? I think we have already added default words to the selectedWords list at this point

**Review by KardelenCeren — CHANGES_REQUESTED** [review]

File: src/main/java/app/linguistai/bmvp/service/wordbank/WordSelectionService.java (hunk 3 above)
Comment (KardelenCeren) [reply src/main/java/app/linguistai/bmvp/service/wordbank/WordSelectionService.java:156]: @tolgaozgun

**Review by KardelenCeren — APPROVED** [review]
Looks great!

**Review by can-aslan — APPROVED** [review]

**Pull request merged**